### PR TITLE
Add a devcontainer for gpu dev

### PR DESCRIPTION
**Pull request by JackCaoG**
I think we should have a 3.10 dev image too. 3.8 is too old.

### CODE REVIEW

File: docs/gpu.md [hunk 1]
```diff
@@ -64,8 +64,8 @@ Thu Dec  8 06:24:29 2022
 Make sure `PATH` and `LD_LIBRARY_PATH` environment variables account for cuda. Please do a `echo $PATH` and `echo $LD_LIBRARY_PATH` to verify. If not, please follow [link](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#mandatory-actions) to do so. Example:
 
 ```
-echo "export PATH=/usr/local/cuda-12.1/bin${PATH:+:${PATH}}" >> ~/.bashrc
-echo "export LD_LIBRARY_PATH=/usr/local/cuda-12.1/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}" >> ~/.bashrc
+echo "export PATH=\$PATH:/usr/local/cuda-12.1/bin" >> ~/.bashrc
+echo "export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:/usr/local/cuda-12.1/lib64" >> ~/.bashrc
```
Comment (JackCaoG) [inline docs/gpu.md:68]: the old way of export PATH will overwrite the vscode PATH and I can;t use `code` to open files in the vscode. I tested above command works.

**Review by will-cromar — APPROVED** [review]
Yeah, 3.8 is old. Our CUDA 12.3 image has 3.10

**Comment by JackCaoG** [comment]
TPU build failed with
```
TASK [build_srcs : Build PyTorch] **********************************************
Friday 12 April 2024  23:27:56 +0000 (0:00:00.255)       0:00:01.561 ********** 
fatal: [localhost]: FAILED! => changed=true 
  cmd:
  - python
  - setup.py
  - bdist_wheel
  delta: '0:00:00.447316'
  end: '2024-04-12 23:27:56.8[68](https://github.com/pytorch/xla/actions/runs/8669115427/job/23775314658?pr=6923#step:6:69)795'
  msg: non-zero return code
  rc: 1
  start: '2024-04-12 23:27:56.421479'
  stderr: ''
  stderr_lines: <omitted>
  stdout: |-
    Building wheel torch-2.4.0
    -- Building version 2.4.0
    Could not find any of CMakeLists.txt, Makefile, setup.py, LICENSE, LICENSE.md, LICENSE.txt in /__w/xla/xla/pytorch/third_party/pybind11
    Did you run 'git submodule update --init --recursive'?
  stdout_lines: <omitted>
```
It doesn't seems like it is related, @will-cromar can you confirm?

**Comment by will-cromar** [comment]
Yeah, I was just looking at that. The CI doesn't use the devcontainer config at all. No idea why the torch build is complaining about pybind. The logs show that it's there:

```
/usr/bin/git submodule status
...
-3e9dfa2866941655c56877882565e7577de6fc7b third_party/pybind11
...
```

You should be good to merge.

**Pull request merged**